### PR TITLE
find: Implement support for multiple directories

### PR DESCRIPTION
**Pull request by timschumi**
The changes include a port to LibMain (since I originally planned to use the `strings` Span for parsing, and now there is no point in removing it again) and a bit of refactoring that improves parsing, since `getopt()` didn't really understand options that came after a non-option string (so now we are using a Vector of arguments that we constantly pop values from and just parse that part manually).

### CODE REVIEW

File: Userland/Utilities/find.cpp [hunk 1]
```diff
@@ -567,7 +537,31 @@ static void walk_tree(FileData& root_data, Command& command)
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    LexicalPath root_path(parse_options(arguments.argc, arguments.argv));
+    Vector<char*> args;
+    args.ensure_capacity(arguments.argc);
+    for (int i = 1; i < arguments.argc; ++i)
+        args.unchecked_append(arguments.argv[i]);
```
Comment (alimpfard) [inline Userland/Utilities/find.cpp:543]: Maybe
```c++
auto args = arguments.strings;
args.take_first();
```
Instead?

Comment (alimpfard) [reply Userland/Utilities/find.cpp:543]: Oh it's a bunch of `char*`s
Then how about
```c++
Vector<char*> args;
args.append(arguments.argv + 1, arguments.argc - 1);
```
?

Comment (timschumi) [reply Userland/Utilities/find.cpp:543]: I was looking at what LibMain is doing (the code that is currently in the PR is basically a copy of that with changed bounds and without the Span conversion), but if what you suggest works that would be even better.

File: Userland/Utilities/find.cpp [hunk 2]
```diff
@@ -563,17 +565,13 @@ static void walk_tree(FileData& root_data, Command& command)
...
+    LexicalPath root_path(parse_options(arguments.argc, arguments.argv));
     String dirname = root_path.dirname();
     String basename = root_path.basename();
 
-    int dirfd = open(dirname.characters(), O_RDONLY | O_DIRECTORY | O_CLOEXEC);
-    if (dirfd < 0) {
-        perror(dirname.characters());
-        return 1;
-    }
+    int dirfd = TRY(Core::System::open(dirname.characters(), O_RDONLY | O_DIRECTORY | O_CLOEXEC));
```
Comment (awesomekling) [inline Userland/Utilities/find.cpp]: You can drop the `.characters()` here, as `Core::System::open()` takes a `StringView` :)

Comment (timschumi) [reply Userland/Utilities/find.cpp]: Very cool, haven't even thought of checking if the signature for the wrapper changed!

**Comment by timschumi** [comment]
The build failure seems to be the usual again, so this PR should be ready.

**Comment by AtkinsSJ** [comment]
Nice! It would be good to update the man page too to mention that multiple paths are supported. :^)

**Comment by timschumi** [comment]
> Nice! It would be good to update the man page too to mention that multiple paths are supported. :^)

Sounds good, will do!

> 
> EDIT: Also, noticed you marked Andreas's feedback as resolved but forgot to actually make the suggested change. ;)

It seems like I rebased a wrong/outdated branch, since the `.characters()` was actually gone in  d6daaaa. Will fix.

**Pull request merged**